src/pages/SwapNew/components/CurrencyItem/CurrencyAmount.tsx

```diff
@@ -0,0 +1,104 @@
+import { CurrencyAmount as CurrencyAmountType } from '@swapr/sdk'
+
+import { ChangeEvent, useState, useMemo, useCallback, useEffect } from 'react'
+import styled from 'styled-components'
+
+import { escapeRegExp, normalizeInputValue } from '../../../../utils'
+import { debounce } from '../../../../utils/debounce'
+import { TEXT_COLOR_PRIMARY } from '../../constants'
+import { FontFamily } from '../../constants'
+
+const inputRegex = RegExp(`^\\d*(?:\\\\[.])?\\d*$`)
+
+type CurrencyAmountProps = {
+  value: string | number
+  onUserInput: (value: string) => void
+  setIsMaxAmount: (value: boolean) => void
+  maxAmount?: CurrencyAmountType
+  isInputPanelDisabled?: boolean
+}
+
+export function CurrencyAmount({
+  value,
+  onUserInput,
+  setIsMaxAmount,
+  maxAmount,
+  isInputPanelDisabled,
+}: CurrencyAmountProps) {
+  const [inputValue, setInputValue] = useState(value)
+
+  const handleOnUserInput = useCallback(
+    (value: string) => {
+      if (maxAmount?.toExact() === value) setIsMaxAmount(true)
+      else setIsMaxAmount(false)
+
+      onUserInput(value)
+    },
+    [maxAmount, onUserInput, setIsMaxAmount]
+  )
+
+  const debouncedUserInput = useMemo(() => {
+    return debounce(handleOnUserInput, 250)
+  }, [handleOnUserInput])
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    event.stopPropagation()
+
+    const parsedValue = event.target.value.replace(/,/g, '.')
+
+    if (parsedValue === '' || inputRegex.test(escapeRegExp(parsedValue))) {
+      setInputValue(normalizeInputValue(parsedValue))
+      debouncedUserInput(normalizeInputValue(parsedValue))
+    }
+  }
+
+  useEffect(() => {
+    if (inputValue !== value) {
+      setInputValue(value)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value])
+
+  return (
+    <StyledInput
+      value={inputValue}
+      onChange={handleInputChange}
+      type="number"
+      inputMode="decimal"
+      autoComplete="off"
+      autoCorrect="off"
+      pattern="^[0-9]*[.,]?[0-9]*$"
+      placeholder="0.0"
+      minLength={1}
+      maxLength={79}
+      spellCheck="false"
+      disabled={isInputPanelDisabled}
+    />
+  )
+}
+
+const StyledInput = styled.input`
+  width: 100%;
+  height: 34px;
+  line-height: 34px;
+  font-size: 28px;
+  ${FontFamily}
```

Comment:
```suggestion
  ${FontFamily}
```

By default I think our font will be `inter` not sure if you want to add again.

Reply:
If the intention is to have the same font on all of the places in the application, then yeah, this is redundant. I'll clarify this.

Reply:
Removed font family styling, I guess we are dependant on base font.